From d29a837011fc6d7088e04d3244ccd477e8e86868 Mon Sep 17 00:00:00 2001
From: Kamal Mostafa <kamal@canonical.com>
Date: Mon, 13 Aug 2012 09:46:35 -0700
Subject: [PATCH] UBUNTU: SAUCE: input: Cypress PS/2 Trackpad fix no-config
 stubs

BugLink: http://launchpad.net/bugs/978807

Provide the no-config stubs in the .h instead of the .c.

Signed-off-by: Kamal Mostafa <kamal@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/input/mouse/cypress_ps2.c | 16 ----------------
 drivers/input/mouse/cypress_ps2.h | 17 +++++++++++++++--
 2 files changed, 15 insertions(+), 18 deletions(-)

diff --git a/drivers/input/mouse/cypress_ps2.c b/drivers/input/mouse/cypress_ps2.c
index 8733676..a370b3a 100644
--- a/drivers/input/mouse/cypress_ps2.c
+++ b/drivers/input/mouse/cypress_ps2.c
@@ -248,8 +248,6 @@ int cypress_detect(struct psmouse *psmouse, bool set_properties)
 	return 0;
 }
 
-#ifdef CONFIG_MOUSE_PS2_CYPRESS
-
 static int cypress_read_fw_version(struct psmouse *psmouse)
 {
 	struct cytp_data *cytp = psmouse->private;
@@ -941,17 +939,3 @@ bool cypress_supported(void)
 {
 	return true;
 }
-
-#else  /* CONFIG_MOUSE_PS2_CYPRESS */
-
-int cypress_init(struct psmouse *psmouse)
-{
-	return -ENOSYS;
-}
-
-bool cypress_supported(void)
-{
-	return false;
-}
-
-#endif	/* CONFIG_MOUSE_PS2_CYPRESS */
diff --git a/drivers/input/mouse/cypress_ps2.h b/drivers/input/mouse/cypress_ps2.h
index de4aa58..dd38658 100644
--- a/drivers/input/mouse/cypress_ps2.h
+++ b/drivers/input/mouse/cypress_ps2.h
@@ -3,8 +3,6 @@
 
 #include "psmouse.h"
 
-#define CONFIG_MOUSE_PS2_CYPRESS
-
 #define CMD_BITS_MASK 0x03
 #define COMPOSIT(x, s) (((x) & CMD_BITS_MASK) << (s))
 
@@ -200,8 +198,23 @@ struct cytp_data {
 };
 
 
+#ifdef CONFIG_MOUSE_PS2_CYPRESS
 int cypress_detect(struct psmouse *psmouse, bool set_properties);
 int cypress_init(struct psmouse *psmouse);
 bool cypress_supported(void);
+#else
+inline int cypress_detect(struct psmouse *psmouse, bool set_properties)
+{
+	return -ENOSYS;
+}
+inline int cypress_init(struct psmouse *psmouse)
+{
+	return -ENOSYS;
+}
+inline bool cypress_supported(void)
+{
+	return 0;
+}
+#endif /* CONFIG_MOUSE_PS2_CYPRESS */
 
 #endif  /* _CYPRESS_PS2_H */
-- 
1.8.0

